internal/types/kruizePayload/common.go

```diff
@@ -91,103 +91,160 @@ func convertMetricToString(data interface{}) string {
 	if metric, ok := data.(float64); ok {
 		return fmt.Sprintf("%.2f", metric)
 	} else {
-		return "-1"
+		return ""
 	}
 }
 
 func make_container_data(c map[string]interface{}) container {
-	container_data := container{
-		Container_image_name: c["image_name"].(string),
-		Container_name:       c["container_name"].(string),
-		Metrics: []metric{
-			{
-				Name: "cpuRequest",
-				Results: result{
-					Aggregation_info: aggregation_info{
-						Sum:    convertMetricToString(c["cpu_request_container_sum_SUM"]),
-						Avg:    convertMetricToString(c["cpu_request_container_avg_MEAN"]),
-						Format: "cores",
-					},
+
+	metrics := []metric{}
+
+	// cpuRequest
+	sum := convertMetricToString(c["cpu_request_container_sum_SUM"])
+	avg := convertMetricToString(c["cpu_request_container_avg_MEAN"])
+	if sum != "" && avg != "" {
+		metrics = append(metrics, metric{
+			Name: "cpuRequest",
+			Results: result{
+				Aggregation_info: aggregation_info{
+					Sum:    sum,
+					Avg:    avg,
+					Format: "cores",
 				},
 			},
-			{
-				Name: "cpuLimit",
-				Results: result{
-					Aggregation_info: aggregation_info{
-						Sum:    convertMetricToString(c["cpu_limit_container_sum_SUM"]),
-						Avg:    convertMetricToString(c["cpu_limit_container_avg_MEAN"]),
-						Format: "cores",
-					},
+		})
+	}
+
+	// cpuLimit
+	sum = convertMetricToString(c["cpu_limit_container_sum_SUM"])
+	avg = convertMetricToString(c["cpu_limit_container_avg_MEAN"])
+	if sum != "" && avg != "" {
+		metrics = append(metrics, metric{
+			Name: "cpuLimit",
+			Results: result{
+				Aggregation_info: aggregation_info{
+					Sum:    sum,
+					Avg:    avg,
+					Format: "cores",
 				},
 			},
-			{
-				Name: "cpuUsage",
-				Results: result{
-					Aggregation_info: aggregation_info{
-						Min:    convertMetricToString(c["cpu_usage_container_min_MIN"]),
-						Max:    convertMetricToString(c["cpu_usage_container_max_MAX"]),
-						Sum:    convertMetricToString(c["cpu_usage_container_sum_SUM"]),
-						Avg:    convertMetricToString(c["cpu_usage_container_avg_MEAN"]),
-						Format: "cores",
-					},
+		})
+	}
+
+	// cpuUsage
+	sum = convertMetricToString(c["cpu_usage_container_sum_SUM"])
+	avg = convertMetricToString(c["cpu_usage_container_avg_MEAN"])
+	max := convertMetricToString(c["cpu_usage_container_max_MAX"])
+	min := convertMetricToString(c["cpu_usage_container_min_MIN"])
```

Comment:
@patilsuraj767 Shall we check for max and min as well on the if condition for all metric appends?